setup.cfg

```diff
@@ -2,11 +2,16 @@
 addopts = -v --junit-xml=test-report.xml
           --doctest-modules
           --cov=riotctrl --cov-branch
-          --cov-report=term --cov-report=xml --cov-report=html
+          --cov-report=term-missing --cov-report=xml
 testpaths = riotctrl
 markers =
     rapidjson
 
+[coverage:report]
+exclude_lines = pragma: no cover
+                raise NotImplementedError
+                return NotImplemented
```

Comment:
Why do need to add these lines ? In theory, these kind of things could be tested.

Reply:
> Why do need to add these lines ? In theory, these kind of things could be tested.

I was called Mr "100% percent coverage" by @MrKevinWeiss before, but even I don't test code that states on its own that it is not implemented ;-).

Reply:
I think this is a bit out of scope of this PR anyway. And I think, this only concerns https://github.com/RIOT-OS/riotctrl/blob/ee12bb524ae79da7f34428be9743ac1c638dc506/riotctrl/ctrl.py#L237
The "raise NotImplementedError" is useless since it's an abstract method, you could just replace it with `...`. When instanciating a concrete class that doesn't implement the  `get_ctrl` method, python will raise a TypeError exception telling what method is missing.

Reply:
To summarize, I think this whole should be removed.

Reply:
> I think this is a bit out of scope of this PR anyway. And I think, this only concerns
> 
> https://github.com/RIOT-OS/riotctrl/blob/ee12bb524ae79da7f34428be9743ac1c638dc506/riotctrl/ctrl.py#L237

But then get a missing coverage there....

Reply:
> then get a missing coverage there...

Then just replace this line with `...  # pragma: no cover`

If you want to ensure a derived class cannot be instanciated without a proper implementation of the `get_ctrl` method, you can use this test:

```py
def test_not_implemented_factory():
    class MyFactory(riotctrl.ctrl.RIOTCtrlFactoryBase):
        pass

    with pytest.raises(TypeError) as exc_info:
        _ = MyFactory()

    assert exc_info.value.args[0] == (
        "Can't instantiate abstract class MyFactory "
        "with abstract method get_ctrl"
    )
```

but you definitely don't need to add an `exclude_lines` config in setup.cfg.

Reply:
Fine... let's finally get this in and worry about actual coverage latter...